version: '3.6'

services:
  nextcloud_app:
    depends_on:
      - nextcloud_db
      - traefik
    image: nextcloud
    container_name: nextcloud_app
    restart: unless-stopped
    ports:
      - 83:80
    env_file: .env
    networks:
      - traefik_network
    volumes:
      - ./nextcloud_data:/var/www/html
      - nextcloud-webconf:/etc/apache2
    labels:
      - "traefik.enable=true"
      - "traefik.http.middlewares.myredirect.redirectscheme.scheme=https"
      - "traefik.http.routers.nextcloud_app.middlewares=myredirect"
      - "traefik.http.routers.nextcloud_app.rule=Host(`homelab.nextcloud.local`)"
      - "traefik.http.routers.nextcloud_app.entrypoints=web"
      - "traefik.http.routers.nextcloud_app-secure.rule=Host(`homelab.nextcloud.local`)"
      - "traefik.http.routers.nextcloud_app-secure.entrypoints=websecure"
      - "traefik.http.routers.nextcloud_app-secure.tls.certresolver=myhttpchallenge"
      - "traefik.http.routers.nextcloud_app-secure.tls=true"

  nextcloud_db:
    image: postgres
    container_name: nextcloud_db
    env_file: .env
    volumes:
      - ./postgres_data:/var/lib/postgresql/data

networks:
  traefik_network:
    external: true

volumes:
  nextcloud-html:
  nextcloud-webconf:
